crates/avalanche-types/src/warp/mod.rs

```diff
@@ -0,0 +1,33 @@
+pub mod client;
+
+use crate::proto::warp::SignResponse;
+use std::io::Result;
+
+#[tonic::async_trait]
+pub trait WarpSignerClient_: Send + Sync + CloneBox {
+    async fn sign(
+        &self,
+        network_id: u32,
+        source_chain_id: &str,
+        payload: &[u8],
+    ) -> Result<SignResponse>;
+}
+
+pub trait CloneBox {
+    fn clone_box(&self) -> Box<dyn WarpSignerClient_ + Send + Sync>;
+}
+
+impl<T> CloneBox for T
+where
+    T: 'static + WarpSignerClient_ + Clone + Send + Sync,
+{
+    fn clone_box(&self) -> Box<dyn WarpSignerClient_ + Send + Sync> {
+        Box::new(self.clone())
+    }
+}
+
+impl Clone for Box<dyn WarpSignerClient_ + Send + Sync> {
+    fn clone(&self) -> Box<dyn WarpSignerClient_ + Send + Sync> {
+        self.clone_box()
+    }
+}
```

Comment:
None of these are necessary:
https://doc.rust-lang.org/stable/std/boxed/struct.Box.html#impl-Clone-for-Box%3CT,+A%3E

Why do you need the trait object to be `Clone` in the first place?